pgmanage/app/client_manager/client_manager.py

```diff
@@ -0,0 +1,544 @@
+            client_id=client_id,
+            conn_tab_id=conn_tab_id,
+            tab=tab,
+            tab_id=tab_id,
+            is_main_tab=False,
+        )
+        return tab
+
+    def create_main_tab(
+        self, client_id: str, conn_tab_id: str, tab: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Creates a new main tab with the given details and
+        adds it to the client's connection sessions.
+
+        Args:
+            client_id (str): The ID of the client.
+            conn_tab_id (str): The ID of the connection tab.
+            tab (Dict[str, Any]): The details of the main tab.
+
+        Returns:
+            Dict[str, Any]: The created main tab object.
+        """
+        tab = self._create_tab_internal(
+            client_id=client_id, conn_tab_id=conn_tab_id, tab=tab, is_main_tab=True
+        )
+        return tab
+
+    def _create_tab_internal(
+        self,
+        client_id: str,
+        conn_tab_id: str,
+        tab: Dict[str, Any],
+        tab_id: Optional[str] = None,
+        is_main_tab: bool = False,
+    ) -> Dict[str, Any]:
+        """Internal method to create a new tab with the given details and
+        add it to the client's connection sessions.
+
+        Args:
+            client_id (str): The ID of the client.
+            conn_tab_id (str): The ID of the connection tab.
+            tab (Dict[str, Any]): The details of the tab.
+            tab_id (Optional[str], optional): The ID of the tab. Defaults to None.
+            is_main_tab (bool, optional): Specifies if the tab is a main tab. Defaults to False.
+
+        Returns:
+            Dict[str, Any]: The created tab object.
+        """
+
+        tab["last_update"] = datetime.now()
+        tab["to_be_removed"] = False
+
+        client = self.get_or_create_client(client_id=client_id)
+
+        if is_main_tab:
+            client["connection_sessions"][conn_tab_id] = tab
+        else:
+            client["connection_sessions"][conn_tab_id]["tab_list"][tab_id] = tab
+
+        return tab
+
+    def close_tab(
+        self, client_id: str, conn_tab_id: str, tab_id: Optional[str] = None
+    ) -> None:
+        """Closes the tab with the given IDs by removing it from the client's connection sessions
+        and closing associated connections.
+
+        Args:
+            client_id (str): The ID of the client.
+            conn_tab_id (str): The ID of the connection tab.
+            tab_id (Optional[str], optional): The ID of the tab. Defaults to None.
+
+        Returns:
+            None
+        """
+
+        client = self.get_or_create_client(client_id=client_id)
+        tab = self.get_tab(client_id=client_id, tab_id=tab_id, conn_tab_id=conn_tab_id)
+        if tab is None:
+            return
+
+        if tab_id:
+            del client["connection_sessions"][conn_tab_id]["tab_list"][tab_id]
+        else:
+            del client["connection_sessions"][conn_tab_id]
+
+        if (
```

Comment:
maybe replace with smth like
if tab["type"] in ["query", "console", "connection", "edit"] ?

Reply:
Done.